src/test/portmidicontroller_test.cpp

```diff
@@ -0,0 +1,480 @@
+            .WillOnce(Return(pmNoError));
+    EXPECT_CALL(*m_mockInput, isOpen())
+            .InSequence(input)
+            .WillRepeatedly(Return(true));
+    EXPECT_CALL(*m_mockInput, close())
+            .InSequence(input)
+            .WillOnce(Return(pmNoError));
+
+    Sequence output;
+    ON_CALL(*m_mockOutput, isOpen())
+            .WillByDefault(Return(false));
+    EXPECT_CALL(*m_mockOutput, openOutput())
+            .WillOnce(Return(pmNoError));
+    EXPECT_CALL(*m_mockOutput, isOpen())
+            .InSequence(output)
+            .WillRepeatedly(Return(true));
+    EXPECT_CALL(*m_mockOutput, close())
+            .InSequence(output)
+            .WillOnce(Return(pmNoError));
+
+    openDevice();
+    EXPECT_TRUE(m_pController->isOpen());
+    closeDevice();
+    EXPECT_FALSE(m_pController->isOpen());
+};
+
+TEST_F(PortMidiControllerTest, WriteShort) {
+    Sequence output;
+    EXPECT_CALL(*m_mockOutput, isOpen())
+            .WillRepeatedly(Return(true));
+    EXPECT_CALL(*m_mockOutput, writeShort(0x403C90))
+            .InSequence(output)
+            .WillOnce(Return(pmNoError));
+    EXPECT_CALL(*m_mockOutput, writeShort(0xFFFFFF))
+            .InSequence(output)
+            .WillOnce(Return(pmBadData));
+    EXPECT_CALL(*m_mockOutput, writeShort(0x403C80))
+            .InSequence(output)
+            .WillOnce(Return(pmNoError));
+
+    m_pController->sendShortMsg(0x90, 0x3C, 0x40);
+    m_pController->sendShortMsg(0xFF, 0xFF, 0xFF);
+    m_pController->sendShortMsg(0x80, 0x3C, 0x40);
+};
+
+TEST_F(PortMidiControllerTest, WriteSysex) {
+    QList<int> sysex;
+    sysex.append(0xF0);
+    sysex.append(0x12);
+    sysex.append(0xF7);
+
+    EXPECT_CALL(*m_mockOutput, isOpen())
+            .WillRepeatedly(Return(true));
+    EXPECT_CALL(*m_mockOutput, writeSysEx(ByteArrayEquals(sysex)))
+            .WillOnce(Return(pmNoError));
+    m_pController->sendSysexMsg(sysex, sysex.length());
+};
+
+TEST_F(PortMidiControllerTest, WriteSysex_Malformed) {
+    QList<int> sysex;
+    sysex.append(0xF0);
+    sysex.append(0x12);
+
+    EXPECT_CALL(*m_mockOutput, isOpen())
+            .WillRepeatedly(Return(true));
+    EXPECT_CALL(*m_mockOutput, writeSysEx(_))
+            .Times(0);
+    m_pController->sendSysexMsg(sysex, sysex.length());
+};
+
+TEST_F(PortMidiControllerTest, Poll_Read_NoInput) {
+    Sequence poll;
+    EXPECT_CALL(*m_mockInput, isOpen())
+            .WillRepeatedly(Return(true));
+    EXPECT_CALL(*m_mockInput, poll())
+            .InSequence(poll)
+            .WillOnce(Return((PmError)FALSE));
+    EXPECT_CALL(*m_mockInput, poll())
+            .InSequence(poll)
+            .WillOnce(Return((PmError)TRUE));
+    EXPECT_CALL(*m_mockInput, read(_, _))
+            .InSequence(poll)
+            .WillOnce(Return(0));
+
+    pollDevice();
+    pollDevice();
+};
+
+TEST_F(PortMidiControllerTest, Poll_Read_Basic) {
+    std::vector<PmEvent> messages;
+    messages.push_back(MakeEvent(0x403C90, 0x0));
```

Comment:
Nope, endianness is transparent to you at this high a level. 

For a C++ integer literal, 0xAABBCC declares that it goes from high (0xAA) to low (0xCC) no matter what the architecture. The compiler deals with putting the bytes in memory in the correct order depending on what architecture you're building for.